analyzer/src/main/kotlin/managers/Carthage.kt

```diff
@@ -0,0 +1,296 @@
+        )
+    }
+
+    private fun parseCarthageDependencies(definitionFile: File): SortedSet<Package> {
+        val dependencyLines = definitionFile.readLines()
+        val workingDir = definitionFile.parent
+        val packages = sortedSetOf<Package>()
+
+        dependencyLines.forEach { line ->
+            if (line.isBlank() && line.isComment()) {
+                return@forEach
+            }
+
+            val (type, id, revision) = parseDependencyLine(line)
+            when (type) {
+                DependencyType.GITHUB -> {
+                    // ID consists of github username/project or a github enterprise URL.
+                    if (isGithubUserAndProject(id)) {
+                        packages.add(createPackageFromGithubUserAndProject(id, revision))
+                    } else {
+                        packages.add(createPackageFromGenericGitUrl(id, revision))
+                    }
+                }
+                DependencyType.GIT -> {
+                    // ID is a generic git URL (https, git or ssh) or a file path to a local repository.
+                    if (isFilePath(workingDir, id)) {
+                        // TODO Carthage supports local git repositories.
+                        //  The downloader should be extended to support `file://` downloads.
+                        packages.add(createPackageFromLocalRepository(id, revision))
+                    } else {
+                        packages.add(createPackageFromGenericGitUrl(id, revision))
+                    }
+                }
+                DependencyType.BINARY -> {
+                    // ID is an URL or a path to a file that contains a Carthage binary project specification.
+                    if (isFilePath(workingDir, id)) {
+                        val filePath = id.removePrefix("file://")
+                        val binarySpecFile = when {
+                            isAbsolutePath(filePath) -> File(filePath)
+                            else -> File("$workingDir/$filePath")
+                        }
+                        val content = binarySpecFile.readText()
+                        val binarySpec = jsonMapper.readValue<Map<String, String>>(content)
+                        packages.add(createPackageFromBinarySpec(binarySpec, id, revision))
+                    } else {
+                        val content = URL(id).readText()
+                        val binarySpec = jsonMapper.readValue<Map<String, String>>(content)
+                        packages.add(createPackageFromBinarySpec(binarySpec, id, revision))
+                    }
+                }
+            }
+        }
+
+        return packages
+    }
+
+    private fun parseDependencyLine(line: String): DependencyDefinition {
+        val split = line.split(" ", limit = 3)
+        val type = DependencyType.valueOf(split[0].toUpperCase())
+        val id = split[1].removeSurrounding("\"")
+        val revision = split[2].removeSurrounding("\"")
```

Comment:
Just a suggestion: You might want to move the above three to named arguments for `DependencyDefinition` similar to like you do for `VcsInfo` below.